### DISCUSSION TITLE
the problem for "conda '/path/to/an/existing/env/directory'"

### DISCUSSION THREAD

**Discussion by Lola58** [nextflow-io/nextflow#4774] [Q&A]
The version of nextflow is
N E X T F L O W ~ version 23.10.1
However, according to the official website command:

nextflow
Copy code
process foo {
  conda '/path/to/an/existing/env/directory'

  '''
  your_command --here
  '''
}
I correctly set the conda environment path in my process, but it did not activate successfully because my command:
python -c "import sys; print(sys.version)" did not display the correct version. However, if I directly set in the script:

bash
Copy code
source /path/to/anaconda3/etc/profile.d/conda.sh
conda activate ${params.condaEnv1} # This variable value is my environment path
it succeeded. Why is this?

**Comment by julia-luz** [comment]
Do you call different conda environments in the same workflow? I believe I may be having a similar issue; the correct conda environment is activated for one process, but not for the next, causing it to fail as the dependencies are wrong.
